schema/encoding/jsonschema/object.go

```diff
@@ -1,15 +1,27 @@
 package jsonschema
 
 import (
-	"io"
+	"fmt"
 
 	"github.com/rs/rest-layer/schema"
 )
 
-func encodeObject(w io.Writer, v *schema.Object) error {
-	ew := errWriter{w: w}
-	if ew.err == nil && v.Schema != nil {
-		ew.err = encodeSchema(w, v.Schema)
+type objectBuilder schema.Object
+
+var (
+	//ErrNoSchema is returned when trying to JSON Encode a schema.Object with the Schema property set to nil.
+	ErrNoSchema = fmt.Errorf("no schema defined for object")
+)
+
+func (v objectBuilder) JSONSchema() (map[string]interface{}, error) {
+	m := make(map[string]interface{})
```

Comment:
Same here, prefer `map[string]interface{}{}`. Only use make when you want to pre-allocate.